$OpenBSD: patch-Makefile,v 1.1.1.1 2019/02/21 00:35:17 jturner Exp $

Use a date stamp for JANET_BUILD to avoid using git.
Replace hardcoded -O2.
Remove hardcoded libraries not on OpenBSD.
Fix installation.

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -27,14 +27,14 @@ PREFIX?=/usr/local
 INCLUDEDIR=$(PREFIX)/include/janet
 LIBDIR=$(PREFIX)/lib
 BINDIR=$(PREFIX)/bin
-JANET_BUILD?="\"$(shell git log --pretty=format:'%h' -n 1)\""
+JANET_BUILD?="\"$(shell date +%Y%m%d)\""
 
-CFLAGS=-std=c99 -Wall -Wextra -Isrc/include -fpic -O2 -fvisibility=hidden \
+CFLAGS=-std=c99 -Wall -Wextra -Isrc/include -fpic $(FLAGS) -fvisibility=hidden \
 	   -DJANET_BUILD=$(JANET_BUILD)
-CLIBS=-lm -ldl
+CLIBS=-lm
 JANET_TARGET=build/janet
 JANET_LIBRARY=build/libjanet.so
-JANET_PATH?=/usr/local/lib/janet
+JANET_PATH?=$(LIBDIR)/janet
 DEBUGGER=gdb
 
 UNAME:=$(shell uname -s)
@@ -42,9 +42,6 @@ LDCONFIG:=ldconfig
 ifeq ($(UNAME), Darwin)
 	# Add other macos/clang flags
 	LDCONFIG:=
-else
-	CFLAGS:=$(CFLAGS) -rdynamic
-	CLIBS:=$(CLIBS) -lrt
 endif
 
 $(shell mkdir -p build/core build/mainclient build/webclient)
@@ -194,13 +191,11 @@ install: $(JANET_TARGET)
 	mkdir -p $(INCLUDEDIR)
 	cp $(JANET_HEADERS) $(INCLUDEDIR)
 	mkdir -p $(LIBDIR)
-	cp $(JANET_LIBRARY) $(LIBDIR)/libjanet.so
+	cp $(JANET_LIBRARY) $(LIBDIR)
 	mkdir -p $(JANET_PATH)
 	cp tools/cook.janet $(JANET_PATH)
 	cp tools/highlight.janet $(JANET_PATH)
-	cp janet.1 /usr/local/share/man/man1/
-	mandb
-	$(LDCONFIG)
+	cp janet.1 $(PREFIX)/man/man1/
 
 uninstall:
 	-rm $(BINDIR)/../$(JANET_TARGET)
